### PR TITLE
ClientName approved/rejected your bid && Implement a PATCH endpoint for updating about for Users

### PR DESCRIPTION
**Pull request by KAJOiq**

### CODE REVIEW

File: Controllers/Web/v1/UsersController.cs [hunk 1]
```diff
@@ -66,6 +68,26 @@ public async Task<IActionResult> GetProfileByIdAsync([FromRoute]long id)
             return NotFound(CreateErrorResponse(StatusCodes.Status404NotFound.ToString(), "NotFound"));
 
         }
+
+        // I need your opinion on updating about for user :)
```
Comment (MohammedLilo) [inline Controllers/Web/v1/UsersController.cs:72]: This controller is the best choice.

File: Controllers/Web/v1/UsersController.cs [hunk 2]
```diff
@@ -66,6 +68,26 @@ public async Task<IActionResult> GetProfileByIdAsync([FromRoute]long id)
...
+        // Add it here? Or add it to every controller?
+        [HttpPatch("about")]
+        public async Task<IActionResult> UpdateAboutAsync([FromBody] UpdateUsersAboutDTO updateUsersAboutDTO)
+        {
+            if (!ModelState.IsValid)
+                return CustomBadRequest();
+
+            var identity = HttpContext.User.Identity as ClaimsIdentity;
+            long authenticatedUserId = authService.GetUserId(identity);
```
Comment (MohammedLilo) [inline Controllers/Web/v1/UsersController.cs:81]: getting the userId from authService.GetUserId, is not useful here, instead you can use `userManager.GetUserAsync(HttpContext.User)`.

File: Controllers/Web/v1/UsersController.cs [hunk 3]
```diff
@@ -66,6 +68,26 @@ public async Task<IActionResult> GetProfileByIdAsync([FromRoute]long id)
...
+
+            var user = await mainAppContext.Users.FindAsync(authenticatedUserId);
+            if (user == null)
+                return NotFound(CreateErrorResponse(StatusCodes.Status404NotFound.ToString(), "User not found"));
+
+            user.About = updateUsersAboutDTO.About;
+            await mainAppContext.SaveChangesAsync();
+
+            return Ok(CreateSuccessResponse(new { Message = "Users About section updated successfully" }));
```
Comment (MohammedLilo) [inline Controllers/Web/v1/UsersController.cs:90]: Please replace this line with `return Ok(CreateSuccessResponse("Users About section updated successfully"));`  as we are using this scheme for all similar responses.

File: Models/DTOs/UpdateUsersAboutDTO.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+﻿namespace AonFreelancing.Models.DTOs
+{
+    public class UpdateUsersAboutDTO
```
Comment (MohammedLilo) [inline Models/DTOs/UpdateUsersAboutDTO.cs:3]: Avoid using verbs in class names, use nouns instead. For example, `UserAboutInputDTO`

**Review by MohammedLilo — COMMENTED** [review]
- Just check out my comments. 
- Should be tested during the meeting and merged today.

File: Controllers/Web/v1/UsersController.cs (hunk 1 above)
Comment (KAJOiq) [reply Controllers/Web/v1/UsersController.cs:72]: Nice

File: Models/DTOs/UpdateUsersAboutDTO.cs (hunk 4 above)
Comment (KAJOiq) [reply Models/DTOs/UpdateUsersAboutDTO.cs:3]: Done

File: Controllers/Web/v1/UsersController.cs (hunk 3 above)
Comment (KAJOiq) [reply Controllers/Web/v1/UsersController.cs:90]: By mistake

File: Controllers/Web/v1/UsersController.cs (hunk 2 above)
Comment (KAJOiq) [reply Controllers/Web/v1/UsersController.cs:81]: we don't have UserService :)

Comment (KAJOiq) [reply Controllers/Web/v1/UsersController.cs]: Done

**Review by MohammedLilo — APPROVED** [review]

**Pull request merged**